src/services/review/ReviewRequestService.ts

```diff
@@ -71,70 +100,163 @@ export default class ReviewRequestService {
+          )
+        )
+      )
+
+  private compareDiffWithMappings = (
+    sessionData: UserWithSiteSessionData,
+    stagingLink: StagingPermalink,
+    files: RawFileChangeInfo[],
+    mappings: Record<
+      string,
+      {
+        author: string
+        unixTime: number
+      }
+    >
+  ): ResultAsync<WithEditMeta<EditedItemDto>, BaseEditedItemDto>[] => {
     // Step 1: Get the site name
     const { siteName } = sessionData
 
-    // Step 2: Get the list of changed files using Github's API
-    // Refer here for details; https://docs.github.com/en/rest/commits/commits#compare-two-commits
-    // Note that we need a triple dot (...) between base and head refs
-    const { files, commits } = await this.apiService.getCommitDiff(siteName)
+    return files.map(({ filename, contents_url: contents }) => {
+      const extractedPathResult = extractPathInfo(filename)
+      if (extractedPathResult.isErr()) {
+        return errAsync({
+          name: "",
+          path: [],
+          type: "file",
+        })
+      }
 
-    const mappings = await this.computeShaMappings(commits)
+      const pathInfo = extractedPathResult.value
 
-    return Promise.all(
-      files.map(async ({ filename, contents_url }) => {
-        const fullPath = filename.split("/")
-        const items = contents_url.split("?ref=")
-        // NOTE: We have to compute sha this way rather than
-        // taking the file sha.
-        // This is because the sha present on the file is
-        // a checksum of the files contents.
-        // And the actual commit sha is given by the ref param
-        const sha = items[items.length - 1]
-        const pageNameRes = this.pageService.parsePageName(
-          filename,
-          sessionData
+      return this.extractConfigInfo(pathInfo)
+        .orElse(() => this.extractMediaInfo(pathInfo))
+        .asyncMap<EditedItemDto>(async (item) => item)
+        .orElse(() =>
+          this.extractPageInfo(pathInfo, sessionData, stagingLink, siteName)
         )
-        const stagingUrl = await pageNameRes
-          .andThen((pageName) =>
-            this.pageService.retrieveStagingPermalink(
-              sessionData,
-              stagingLink,
-              pageName
-            )
-          )
-          // NOTE: We ignore the errors and use a placeholder
-          .unwrapOr("")
-        const fileUrl = await pageNameRes
-          .andThen((pageName) =>
-            this.pageService.retrieveCmsPermalink(pageName, siteName)
-          )
-          .unwrapOr("")
+        .map<WithEditMeta<EditedItemDto>>((item) => {
+          const items = contents.split("?ref=")
+          // NOTE: We have to compute sha this way rather than
+          // taking the file sha.
+          // This is because the sha present on the file is
+          // a checksum of the files contents.
+          // And the actual commit sha is given by the ref param
+          const hash = items[items.length - 1]
```

Comment:
shld we do a length check first?

is there a possibility ref= doesn't exist? if so, that would be an error state right (thought it technically wont break the code)

Reply:
re: this - the github schema [here](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits) has a `?ref=` at the end, which is why i made this assumption

i think it's possible that it doesn't exist but it does imply that github made a breaking change to their api

Reply:
wait based on the schema, why are we not taking the `base_commit`.`sha` directly ah?

Reply:
sorry, where did you see this? i saw mine under `Comparing two commits`, where the schema explicitly has a `?ref=` at the end (returned by github)

Reply:
@kishore03109 i've a comment there (up above) and it's cos the sha present is a checksum of the files' contents but we want the sha referencing the actual file